src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+        if (request.spatial === SpatialMethod.Box.value) {
+            merge(tReq, {'size': conesize});
+        }else{
+            merge(tReq, {'radius': conesize});
+        }
+    }
+
+    if (request.spatial === SpatialMethod.Polygon.value) {
+        const p = escape(request.polygoncoords);//encodeURI
+        encodeURI(request.polygoncoords);
+        merge(tReq, {'polygon': p});
+    }
+
+    dispatchTableSearch(tReq);
+}
+
+function doVoSearch(request) {
+    var tReq = TableRequest.newInstance({
+        [ServerParams.USER_TARGET_WORLD_PT]: request[ServerParams.USER_TARGET_WORLD_PT],
+        id: 'GatorQuery',
+        title: request.catalog,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        radius: request.conesize,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+    });
+    console.log('Does not dispatch yet ' + tReq);
+    //dispatchTableSearch(tReq);
+}
+
+function doLoadTable(request) {
+    var tReq = TableRequest.newInstance({
+        id: 'GatorQuery',
+        title: request.catalog,
+        SearchMethod: request.spatial,
+        catalog: request.cattable,
+        RequestedDataSet: request.catalog,
+        radius: request.conesize,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+    });
+    console.log('Does not dispatch yet ' + tReq);
+    //dispatchTableSearch(tReq);
+}
+
+/**
+ *
+ * @type {ClassicComponentClass<P>}
+ */
+class CatalogSelectView extends Component {
+
+    constructor(props) {
+        super(props);
+        if (!isEmpty(catmaster)) {
+            this.state = {...this.state, master: {catmaster, cols}};
+        } else {
+            this.state = {...this.state, master: {}};
+        }
+    }
+
+    componentWillMount() {
+        if (isEmpty(catmaster)) {
+            this.loadMasterCatalogTable();
+        }
+    }
+
+    /**
+     * Fetch master catalog table with search id = 'irsaCatalogMasterTable'
+     * 'tableModel.tableData.data' returned in the promise is an array of elements which value is refered to the ith column name
+     * from tableModel.tableData.columns
+     * result is set in the state as 'master' object {catmaster, cols}
+     * @see setMaster
+     * @returns {Void} fetch master table and once fetched, set state with it
+     */
+    loadMasterCatalogTable() {
+
+        const request = {id: 'irsaCatalogMasterTable'}; //Fetch master table
+        TblUtil.doFetchTable(request).then((tableModel) => {
+
+            var data = tableModel.tableData.data;
+
+            cols = tableModel.tableData.columns;
+
+            var projects = [], subprojects = {}, option = [];
+            var i;
+            for (i = 0; i < data.length; i++) {
+                if (projects.indexOf(data[i][0]) < 0) {
+                    projects.push(data[i][0]);
+                }
+            }
```

Comment:
I think you could use lodash `uniq` or `uniqBy` with a map and do this loop in one line.